src/DeviceInterface/InterfacePrototype.js

```diff
@@ -799,17 +799,22 @@ class InterfacePrototype {
     postSubmit(values, form) {
         if (!form.form) return;
         if (!form.hasValues(values)) return;
-
-        const isUsernameOnly = Object.keys(values?.credentials || {}).length === 1 && values?.credentials?.username;
-        const checks = [form.shouldPromptToStoreData && !form.submitHandlerExecuted, this.passwordGenerator.generated, isUsernameOnly];
+        const shouldTriggerPartialSave =
+            Object.keys(values?.credentials || {}).length === 1 &&
+            Boolean(values?.credentials?.username) &&
+            Boolean(this.settings.featureToggles.partial_form_saves);
```

Comment:
But I could've missed something, not arguing there 😄

Reply:
Kept `Boolean` in the username part because during debugging `shouldTriggerPartialSave` becomes the actual email string, and has caught me off guard a couple of times. The last one is fair! I am fine relying on JS to do the job there :)

Thanks for the comments.